src/ui/windows/TogglDesktop/TogglDesktop/ui/ViewModels/TimelineViewModel.cs

```diff
@@ -172,21 +171,25 @@ private enum TimeStampType
             End,
             Empty
         }
-        private static List<TimeEntryBlock> ConvertTimeEntriesToBlocks(List<Toggl.TogglTimeEntryView> timeEntries, int selectedScaleMode, DateTime selectedDate)
+        private List<Toggl.TogglTimeEntryView> ConvertTimeEntriesToBlocks(List<Toggl.TogglTimeEntryView> timeEntries, Toggl.TogglTimeEntryView? runningEntry, int selectedScaleMode, DateTime selectedDate)
         {
             var timeStampsList = new List<(TimeStampType Type, TimeEntryBlock Block)>();
             var blocks = new List<TimeEntryBlock>();
             //The idea is to place all the starts and ends in sorted order and then assign an offset to each time entry block from the list:
             // - if it's a start time stamp, then pick up the minimum available offset, if none is available assign a new one.
             // - if it's an end time stamp, then release the offset which it occupied.
-            foreach (var entry in timeEntries)
+            IEnumerable<Toggl.TogglTimeEntryView> allEntries = timeEntries;
+            if (runningEntry != null)
+                allEntries = allEntries.Union(new List<Toggl.TogglTimeEntryView>(){runningEntry.Value});
+            foreach (var entry in allEntries)
             {
                 var startTime = Toggl.DateTimeFromUnix(entry.Started);
-                var height = ConvertTimeIntervalToHeight(startTime, Toggl.DateTimeFromUnix(entry.Ended), selectedScaleMode);
+                var ended = entry.Equals(runningEntry) ? (ulong) Toggl.UnixFromDateTime(DateTime.Now) : entry.Ended;

```

Comment:
Well, I checked that here it doesn't always return `false`, but I can just overload `Equals` method for `TimeEntryView` if this looks suspicious to you. I'd rather don't believe the negative duration, we might not know when else it can be negative.

Reply:
1. You cannot just overload `Equals` for `TimeEntryView` because `TimeEntryView` is auto-generated. In principle, yes, overloading `Equals` for structs sounds good.
2. This is documented that Duration is negative only if TE is running: https://github.com/toggl/toggl_api_docs/blob/master/chapters/time_entries.md#user-content-time-entries:~:text=duration,-%3A%20time. I couldn't find this in Notion's Business Logic document, but we should document it there as well.
Anyway, comparing IDs seems better than comparing the whole struct - a) it's faster, b) it's what we really care about.

It's not a big deal but seems like default struct comparison with reflection is used here - using it is a bad practice in general -and is easily fixable here.